helpers/kubernetes_cluster.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+class Clover
+  def kubernetes_cluster_post(name)
+    authorize("KubernetesCluster:create", @project.id)
+
+    required_parameters = ["name", "location", "kubernetes_version", "private_subnet_id", "cp_nodes", "worker_nodes"]
+    request_body_params = validate_request_params(required_parameters)
+
+    private_subnet = PrivateSubnet.from_ubid(request_body_params["private_subnet_id"])
+
+    unless private_subnet && private_subnet.location == @location
+      if api?
+        fail Validation::ValidationFailed.new({private_subnet_id: "Private subnet with the given id \"#{private_subnet_id}\" and the location \"#{@location}\" is not found"})
+      else
+        flash["error"] = "Private subnet not found"
+        r.redirect "#{@project.path}/kubernetes-cluster"
+      end
```

Comment:
We're returning back with an error message for `Validation::ValidationFailed` on the web. You might remove the branching here if that works for you.

https://github.com/ubicloud/ubicloud/blob/aaff966494396e31772d024f90749fb4732353ad/clover.rb#L225

```suggestion
      fail Validation::ValidationFailed.new({private_subnet_id: "Private subnet with the given id \"#{private_subnet_id}\" and the location \"#{@location}\" is not found"})
```

Reply:
applied. Also, changed all routes to web? only for now.